ast.cpp

```diff
@@ -55,6 +55,18 @@ class ProgramNode : public AstNode {
   std::vector<std::unique_ptr<ExprNode>> exprs_;
 };
 
+class IdExprNode : public ExprNode {
+ public:
+  IdExprNode(std::string id) : id_{id} {}
```

Comment:
```suggestion
  IdExprNode(const std::string& id) : id_{id} {}
```

String copying is considered an expensive operation and is unnecessary in this context.
Prefer passing `std::string`s as const-reference over value.